### PR TITLE
[extractor/hotstar] Add season support

### LINKED ISSUE #5473: Add particular season support for tv show/series for hotstar.com
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a site-specific feature
- [X] I've verified that I'm running yt-dlp version **2022.10.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

India

### Example URLs

https://www.hotstar.com/in/tv/ishqbaaz/9567/seasons/season-2/ss-4357

### Provide a description that is worded well enough to be understood

downloading whole series or show in one go is not possible in which has many episodes, so downloading it by season will help to checkout only that specify season content like many dance, music contest had happened every year so hotstar differentiate them between seasons. so downloading by season is very useful

as you can see below in verbose output whole show as 758 episode which yt-dlp detected successfully but season 2 have total 30 episodes which you can see on this page :- https://www.hotstar.com/in/tv/ishqbaaz/9567/seasons/season-2/ss-4357 

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['https://www.hotstar.com/in/tv/ishqbaaz/9567/seasons/season-2/ss-4357', '-vU', '-s']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.10.04 [4e0511f] (win32_exe)
[debug] Python 3.8.10 (CPython 64bit) - Windows-10-10.0.19044-SP0
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 4.4.1-full_build-www.gyan.dev (setts), ffprobe n5.1-6-gffaf606130-20220809
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.09.24, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Loaded 1690 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.10.04, Current version: 2022.10.04
yt-dlp is up to date (2022.10.04)
[debug] [hotstar:series] Extracting URL: https://www.hotstar.com/in/tv/ishqbaaz/9567/seasons/season-2/ss-4357
[hotstar:series] 9567: Downloading JSON metadata
[hotstar:series] 9567: Downloading JSON metadata
[download] Downloading playlist: 9567
[hotstar:series] Playlist 9567: Downloading 758 videos of 758
[download] Downloading video 1 of 758
[debug] [hotstar] Extracting URL: https://www.hotstar.com/in/tv/ishqbaaz/9567/ignore_me/1000100150
[hotstar] 1000100150: Downloading JSON metadata
[hotstar] 1000100150: Downloading webpage
[hotstar] 1000100150: Downloading token
[hotstar] 1000100150: Downloading JSON metadata
[hotstar] 1000100150: Downloading m3u8 information
[hotstar] 1000100150: Downloading m3u8 information
[hotstar] 1000100150: Downloading MPD manifest
[hotstar] 1000100150: Downloading MPD manifest
[hotstar] 1000100150: Downloading m3u8 information
[hotstar] 1000100150: Downloading MPD manifest
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 1000100150: Downloading 1 format(s): sdr-dash-video_avc1_7+sdr-dash-audio_und_mp4a_3-2
[download] Downloading video 2 of 758
[debug] [hotstar] Extracting URL: https://www.hotstar.com/in/tv/ishqbaaz/9567/ignore_me/1000100272
[hotstar] 1000100272: Downloading JSON metadata
[hotstar] 1000100272: Downloading webpage
[hotstar] 1000100272: Downloading token
[hotstar] 1000100272: Downloading JSON metadata
[hotstar] 1000100272: Downloading m3u8 information
[hotstar] 1000100272: Downloading MPD manifest
[hotstar] 1000100272: Downloading m3u8 information
[hotstar] 1000100272: Downloading MPD manifest
[hotstar] 1000100272: Downloading m3u8 information
[hotstar] 1000100272: Downloading MPD manifest
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 1000100272: Downloading 1 format(s): sdr-dash-video_avc1_7+sdr-dash-audio_und_mp4a_3-2
```

### PR DESCRIPTION
**Pull request by m4tu4g**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

ADD DESCRIPTION HERE

Fixes #5473

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/hotstar.py [hunk 1]
```diff
@@ -312,6 +315,44 @@ def _real_extract(self, url):
...
+        entries = [
+            self.url_result(HotStarIE._video_url(video['contentId'], root=url), HotStarIE, video['contentId'])
+            for video in item_json['items'] if video.get('contentId')]
+
+        return self.playlist_result(entries, season_id)
+
+
 class HotStarSeriesIE(HotStarBaseIE):
     IE_NAME = 'hotstar:series'
     _VALID_URL = r'(?P<url>https?://(?:www\.)?hotstar\.com(?:/in)?/tv/[^/]+/(?P<id>\d+))'
```
Comment (bashonly) [inline yt_dlp/extractor/hotstar.py:358]:
```suggestion
    _VALID_URL = r'(?P<url>https?://(?:www\\.)?hotstar\\.com(?:/in)?/tv/[^/]+/(?P<id>\\d+))(?!/list/)'
```

File: yt_dlp/extractor/hotstar.py [hunk 2]
```diff
@@ -312,9 +315,47 @@ def _real_extract(self, url):
...
+            self.url_result(HotStarIE._video_url(video['contentId'], root=url), HotStarIE, video['contentId'])
+            for video in item_json['items'] if video.get('contentId')]
+
+        return self.playlist_result(entries, season_id)
+
+
 class HotStarSeriesIE(HotStarBaseIE):
     IE_NAME = 'hotstar:series'
-    _VALID_URL = r'(?P<url>https?://(?:www\.)?hotstar\.com(?:/in)?/tv/[^/]+/(?P<id>\d+))'
+    _VALID_URL = r'(?P<url>https?://(?:www\.)?hotstar\.com(?:/in)?/tv/[^/]+/(?P<id>\d+))(?!/list/)'
```
Comment (bashonly) [inline yt_dlp/extractor/hotstar.py:358]:
```suggestion
    _VALID_URL = r'(?P<url>https?://(?:www\.)?hotstar\.com(?:/in)?/tv/[^/]+/(?P<id>\d+))(?!.*/list/)'
```

File: yt_dlp/extractor/hotstar.py [hunk 3]
```diff
@@ -312,9 +315,47 @@ def _real_extract(self, url):
...
+            self.url_result(HotStarIE._video_url(video['contentId'], root=url), HotStarIE, video['contentId'])
+            for video in item_json['items'] if video.get('contentId')]
+
+        return self.playlist_result(entries, season_id)
+
+
 class HotStarSeriesIE(HotStarBaseIE):
     IE_NAME = 'hotstar:series'
-    _VALID_URL = r'(?P<url>https?://(?:www\.)?hotstar\.com(?:/in)?/tv/[^/]+/(?P<id>\d+))'
+    _VALID_URL = r'(?P<url>https?://(?:www\.)?hotstar\.com(?:/in)?/tv/[^/]+/(?P<id>\d+))(?!.*/list/)'
```
Comment (bashonly) [inline yt_dlp/extractor/hotstar.py:358]:
```suggestion
    _VALID_URL = r'(?P<url>https?://(?:www\.)?hotstar\.com(?:/in)?/tv/[^/]+/(?P<id>\d+))/?(?:[#?].*)?$'
```

File: yt_dlp/extractor/hotstar.py [hunk 4]
```diff
@@ -289,7 +289,7 @@ def _real_extract(self, url):
 
 class HotStarPlaylistIE(HotStarBaseIE):
     IE_NAME = 'hotstar:playlist'
-    _VALID_URL = r'https?://(?:www\.)?hotstar\.com/tv/[^/]+/s-\w+/list/[^/]+/t-(?P<id>\w+)'
+    _VALID_URL = r'https?://(?:www\.)?hotstar\.com(?:/in)?/tv/[^/]+/(?:s-)?\w+/list/[^/]+/t-(?P<id>\w+)'
```
Comment (pukkandan) [inline yt_dlp/extractor/hotstar.py:292]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?hotstar\.com(?:/in)?/tv(?:/[^/]+){2}/list/[^/]+/t-(?P<id>\w+)'
```

File: yt_dlp/extractor/hotstar.py [hunk 5]
```diff
@@ -312,9 +315,47 @@ def _real_extract(self, url):
...
+            'x-country-code': 'IN',
+            'x-platform-code': 'PCTV',
+        }
+        item_json = self._download_json(
+            f'{self._API_URL}/o/v1/season/asset?tao=0&tas=0&size=10000&id={season_id}', season_id, headers=headers)['body']['results']
+        entries = [
+            self.url_result(HotStarIE._video_url(video['contentId'], root=url), HotStarIE, video['contentId'])
+            for video in item_json['items'] if video.get('contentId')]
+
+        return self.playlist_result(entries, season_id)
```
Comment (pukkandan) [inline yt_dlp/extractor/hotstar.py:353]: duplicate code

File: yt_dlp/extractor/hotstar.py [hunk 6]
```diff
@@ -312,9 +315,47 @@ def _real_extract(self, url):
...
+
+
 class HotStarSeriesIE(HotStarBaseIE):
     IE_NAME = 'hotstar:series'
-    _VALID_URL = r'(?P<url>https?://(?:www\.)?hotstar\.com(?:/in)?/tv/[^/]+/(?P<id>\d+))'
+    _VALID_URL = r'(?P<url>https?://(?:www\.)?hotstar\.com(?:/in)?/tv/[^/]+/(?P<id>\d+))/?(?:[#?].*)?$'
```
Comment (pukkandan) [inline yt_dlp/extractor/hotstar.py:358]:
```suggestion
    _VALID_URL = r'(?P<url>https?://(?:www\.)?hotstar\.com(?:/in)?/tv/[^/]+/(?P<id>\d+))/?(?:[#?]|$)'
```

**Pull request merged**